modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/service/test/JournalFolderLocalServiceTest.java

```diff
@@ -74,7 +150,49 @@ public void testGetNoAssetFolders() throws Exception {
 		Assert.assertEquals(folders.toString(), folder, folders.get(0));
 	}
 
+	@Test(expected = DuplicateJournalFolderExternalReferenceCodeException.class)
+	public void testJournalFolderWithExistingExternalReferenceCode()
+		throws Exception {
+
+		JournalFolder folder = addJournalFolder(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString());
+
+		addJournalFolder(
+			folder.getExternalReferenceCode(), RandomTestUtil.randomString(),
+			folder.getDescription());
+	}
+
+	@Test
+	public void testUpdateJournalFolderByExternalReferenceCode()
+		throws Exception {
+
+		JournalFolder folder = addJournalFolder(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString());
+
+		JournalFolderLocalServiceUtil.updateFolder(
+			_user.getUserId(), _group.getGroupId(), folder.getFolderId(),
+			folder.getParentFolderId(), folder.getName(),
+			folder.getDescription(), false,
+			ServiceContextTestUtil.getServiceContext(
+				_group, _user.getUserId()));
+	}
+
+	protected JournalFolder addJournalFolder(
```

Comment:
I like this abstraction to avoid duplicate code 🎉

Reply:
+1000